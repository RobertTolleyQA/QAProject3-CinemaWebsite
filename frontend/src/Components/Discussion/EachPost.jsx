import { Toast, ToastBody, ToastHeader } from "reactstrap";
import DeletePost from "./DeletePost";
import EditPost from "./EditPost";

const EachPost = ({ item, trigger }) => {
    return (
        <div className="p-6 my-2 rounded">
            <Toast>
                <ToastHeader><h5>{item.name}'s post about {item.movie}</h5>
                </ToastHeader>
                <ToastBody style={{ height: "auto" }}>
                    <p>{item.topic}...</p>
                    <p>{item.discussion}</p>
                    <p>Rating: {item.rating}</p>
                    <hr />
                    <DeletePost className="float-right" del={item._id} trigger={trigger} />
                    <EditPost className="float-right" item={item} trigger={trigger} />
                </ToastBody>
            </Toast>
        </div>

    )
}
export default EachPost;