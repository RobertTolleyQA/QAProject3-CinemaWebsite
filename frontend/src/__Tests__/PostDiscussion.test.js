//import PostDiscussion from '../Components/Discussion/PostDiscussion';
import { create } from 'react-test-renderer';
import EachPost from '../Components/Discussion/EachPost';


describe(`Post Discussion Testing`, () => {

    const item = {
        name: 'frank',
        movie: 'RED',
        topic: 'Old',
        discussion: 'This is an older film, but it is still really good',
        rating: 7
    }

    let componentToTest;

    beforeEach(() => {
        const TestInstance = create(<EachPost item={item} />);
        componentToTest = TestInstance.root;

    });

    it(`should render a card with the name Frank`, () => {
        const h5 = componentToTest.findByType(`h5`);
        console.log(item.name);
        expect(h5.children[0]).toEqual(item.name);
    });

    it(`should render the movie title of the discussion`, () => {
        const p1 = componentToTest.findAllByType('p')[1];
        console.log(p1.children);
        expect(p1.children).toEqual([item.movie]);
    })

    it(`should render the topic of the discussion`, () => {
        const p3 = componentToTest.findAllByType('p')[3];
        console.log(p3.children);
        expect(p3.children).toEqual([item.topic]);
    })

    it(`should render the discussion of the discussion`, () => {
        const p5 = componentToTest.findAllByType('p')[5];
        console.log(p5.children);
        expect(p5.children).toEqual([item.discussion]);
    })

    it(`should render the rating of the discussion`, () => {
        const p6 = componentToTest.findAllByType('p')[6];
        console.log(p6.children[1]);
        expect(p6.children[1]).toContain([item.rating]);
    })

    it(`should match the snapshot`, () => {
        const instance = create(<EachPost item={item} />).toJSON();
        expect(instance).toMatchSnapshot();
    })


})