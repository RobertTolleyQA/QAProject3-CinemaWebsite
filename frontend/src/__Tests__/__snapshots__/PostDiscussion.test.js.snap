// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post Discussion Testing should match the snapshot 1`] = `
<div
  className="p-6 my-2 rounded"
>
  <div
    className="toast fade"
    id="toast"
    role="alert"
  >
    <div
      className="toast-header"
      id="toastheader"
    >
      <strong
        className="mr-auto"
      >
        <h5>
          frank
          's post about 
          RED
        </h5>
      </strong>
    </div>
    <div
      className="toast-body"
      style={
        Object {
          "height": "auto",
        }
      }
    >
      <p
        id="film-header"
      >
        Film:
      </p>
      <p
        id="film-point"
      >
        RED
      </p>
      <p
        id="topic-header"
      >
        Topic:
      </p>
      <p
        id="topic-point"
      >
        Old
      </p>
      <p
        id="discussion-header"
      >
        Discussion point:
      </p>
      <p
        id="discussion-point"
      >
        This is an older film, but it is still really good
      </p>
      <p
        id="rating"
      >
        Rating: 
        7
      </p>
      <hr />
      <a
        className="btn btn-outline-danger card-link"
        onClick={[Function]}
      >
        X
      </a>
      <a
        className="btn btn-outline-dark card-link"
        onClick={[Function]}
      >
        Edit
      </a>
      <a
        className="btn btn-outline-primary card-link"
      >
        Comment
      </a>
    </div>
  </div>
</div>
`;
